### PR TITLE
feat: Add onboardingStates extension resource to `avm/res/operational-insights/workspace`

### LINKED ISSUE #3340: [AVM Module Issue]: Feature Request-LAW-Need capability to add onboardingStates extension resource
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Issue Type?

Feature Request

### Module Name

avm/res/operational-insights/workspace

### (Optional) Module Version

_No response_

### Description

Can we have the capability to add Microsoft.SecuirtyInsights/onboardingStates (which is an extension resource) to the log analytics workspace resource module. This is primarily a requirement needed for onboarding the log analytics workspace to sentinel.

We currently use [this approach in ALZ-Bicep](https://github.com/Azure/ALZ-Bicep/blob/main/infra-as-code/bicep/modules/logging/logging.bicep#L641C1-L645C2) for reference so it would just need to be an empty resource declaration that we could use to scope to the LAW.

### (Optional) Correlation Id

_No response_

### LINKED ISSUE #3378: [AVM Module Issue]: avm/res/operational-insights/workspace does not support deploy of SQLAuditing solution
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Issue Type?

Bug

### Module Name

avm/res/operations-management/solution

### (Optional) Module Version

_No response_

### Description

Following code fails:

```bicep
module law 'br/public:avm/res/operational-insights/workspace:0.7.0' = {
  name: logAnalyticsWorspaceName
  params: {
    name: logAnalyticsWorspaceName
    enableTelemetry: false
    gallerySolutions: [
      {
        name: 'SQLAuditing'
        product: 'SQLAuditing'
        publisher: 'Microsoft'
      }
    ]
  }
}
```

Error message:

```
Solution product name cannot start with 'OMSGallery/' as it is reserved for Microsoft first party solutions. Operation Id: '8aab36af321b604584069fe60e602148' (Code: InvalidParameter, Target: plan.product)
```

Solution can be added by using solutions resource:

```bicep
var solutionName = 'SQLAuditing(${logAnalyticsWorspaceName})'
resource solution 'Microsoft.OperationsManagement/solutions@2015-11-01-preview' = {
  name: solutionName
  location: location
  properties: {
    workspaceResourceId: law.outputs.resourceId
  }
  plan: {
    name: solutionName
    promotionCode: ''
    product: 'SQLAuditing'
    publisher: 'Microsoft'
  }
}
```

-------

For me it looks like SQLAudit solution is not available on OMSGallery namespace,

https://github.com/Azure/bicep-registry-modules/blob/main/avm/res/operations-management/solution/main.bicep#L48:

```bicep
var solutionProduct = publisher == 'Microsoft' ? 'OMSGallery/${name}' : product
```

--------

For a reference Portal creates following resource(not the call but resulting stete of the resource) when turning on auditing to LAW on Azure SQL:

```json
{
  "plan": {
    "name": "SQLAuditing[law-law]",
    "publisher": "Microsoft",
    "promotionCode": "",
    "product": "SQLAuditing",
    "version": "1.0"
  },
  "properties": {
    "workspaceResourceId": "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/RgName/providers/Microsoft.OperationalInsights/workspaces/law-law",
    "provisioningState": "Succeeded",
    "creationTime": "Thu, 26 Sep 2024 14:55:20 GMT",
    "lastModifiedTime": "Thu, 26 Sep 2024 14:55:20 GMT",
    "containedResources": [
      "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/RgName/providers/Microsoft.OperationalInsights/workspaces/law-law/views/SQLSecurityInsights",
      "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/RgName/providers/Microsoft.OperationalInsights/workspaces/law-law/views/SQLAccessToSensitiveData"
    ],
    "referencedResources": []
  },
  "location": "westeurope",
  "tags": {},
  "id": "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/RgName/providers/Microsoft.OperationsManagement/solutions/SQLAuditing[law-law]",
  "name": "SQLAuditing[law-law]",
  "type": "Microsoft.OperationsManagement/solutions"
}
``` 

### (Optional) Correlation Id

_No response_

### PR DESCRIPTION
**Pull request by krbar**
## Description

- Adds new parameter `onboardWorkspaceToSentinel` controlling the onboardingStates extension resource, which onboards the workspace to Sentinel
- Adds references to the AVM common types
- Updated the cross-reference to the module `avm/res/operations-management/solution`. Now referencing the newest version 0.3.0, which offers more flexibility in the solution naming. This should resolve #3378. 

> Note: this is a breaking change. The format of the expected values of the `gallerySolutions` parameter has changed. Please refer to the updated documentation and parameter descriptions for more information.

Resolves #3340 
Resolves #3378 

## Pipeline Reference

| Pipeline |
| -------- |
|    [avm.res.operational-insights.workspace](https://github.com/krbar/bicep-registry-modules/actions/workflows/avm.res.operational-insights.workspace.yml)      |

## Type of Change

- [ ] Update to CI Environment or utilities (Non-module affecting changes)
- [x] Azure Verified Module updates:
  - [ ] Bugfix containing backwards-compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [x] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [x] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [x] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] I have run `Set-AVMModule` locally to generate the supporting module files.
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings

### CODE REVIEW

**Comment by krbar** [comment]
Keeping this PR in draft status until #3671 is merged and published, so we can reference the newest version of the `avm/res/operations-management/solution` module.

File: avm/res/operational-insights/workspace/main.bicep [hunk 1]
```diff
@@ -496,4 +470,22 @@ type diagnosticSettingType = {
 
   @description('Optional. The full ARM resource ID of the Marketplace resource to which you would like to send Diagnostic Logs.')
   marketplacePartnerResourceId: string?
-}[]?
+}
+
+@export()
+type gallerySolutionType = {
```
Comment (AlexanderSehr) [inline avm/res/operational-insights/workspace/main.bicep:476]: I wonder if we should implement this type in the `solution` module as an exported type and import it all over. Thoughts?

Comment (krbar) [reply avm/res/operational-insights/workspace/main.bicep:476]: @AlexanderSehr thanks for the review! This sounds reasonable. Do you suggest to re-design the parameters of the `solution` module to use this type, or just to create an exportable type for cross-referencing purposes without changing the rest of the module?

Comment (krbar) [reply avm/res/operational-insights/workspace/main.bicep:476]: I created #3718, once it's merged, I will update this PR

Comment (AlexanderSehr) [reply avm/res/operational-insights/workspace/main.bicep:476]: Hey @krbar,
I'd suggest what you did - with the one caveat that I commented in that PR. Thanks :)

Comment (krbar) [reply avm/res/operational-insights/workspace/main.bicep:481]: @AlexanderSehr I've updated the module to use the newest `solution` module and the type definition from there. Please have a look.

**Review by AlexanderSehr — APPROVED** [review]

**Pull request merged**